js/completionCalculations.js

```diff
@@ -110,9 +111,32 @@ define([
         });
     }
 
+    function unavailableInHierarchy(parents) {
+        if (parents.length > 0) {
```

Comment:
Is it actually possible to have a child with no parents?  If so, I think there's a missing `else` and `return` in this code.